src/main/java/aspects/SetBrowserValuesAspect.java

```diff
@@ -0,0 +1,32 @@
+package aspects;
+
+import beans.AuthenticatedRequestBean;
+import beans.SessionNavigationBean;
+import org.aspectj.lang.JoinPoint;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Before;
+import org.javarosa.core.model.utils.DateUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.annotation.Order;
+import services.BrowserValuesProvider;
+
+/**
+ * Created by amstone326 on 1/8/18.
+ */
+@Aspect
+@Order(1)
+public class SetBrowserValuesAspect {
+
+    @Autowired
+    private BrowserValuesProvider browserValuesProvider;
+
+    @Before(value = "@annotation(org.springframework.web.bind.annotation.RequestMapping)")
+    public void setValues(JoinPoint joinPoint) throws Throwable {
+        Object beanArg = joinPoint.getArgs()[0];
```

Comment:
I believe this will crash with `ArrayIndexOutOfBoundsException` when the request doesn't have a body, should check as in https://github.com/dimagi/formplayer/blob/17dc83b30b8ab72c480eaf239f2999c68b20fc91/src/main/java/aspects/MetricsAspect.java#L51-L51

Reply:
thanks